zerver/lib/sessions.py

```diff
@@ -99,3 +104,41 @@ def get_expirable_session_var(
     if delete:
         del session[var_name]
     return value
+
+
+def narrow_request_user(
+    request: HttpRequest, columns: Sequence[str] = (), *, user_id: int | None = None
+) -> UserProfile | AnonymousUser:
+    # In Tornado, on a UserProfile cache miss, we want to not load the
+    # extremely wide default UserProfile select_related.  This helper
+    # will attempt to load the user from memcached, but failing that
+    # will only select the provided columns from the database.
+    # Because this is a narrower column set, we do not fill the
+    # memcached cache from this user, though we do fill the
+    # `request.user` object, and the requester_for_logs.
```

Comment:
Would it be simpler to just have Tornado work with a separate `narrow_user` cache?

Reply:
This might save enough parsing large objects fetched from memcached to significantly impact the runtime of the `GET /events` code path...